src/shared/mount-util.c

```diff
@@ -1856,13 +1856,20 @@ int path_is_network_fs_harder(const char *path) {
         _cleanup_free_ char *fstype = NULL, *options = NULL;
         int r, ret;
 
-        assert(path);
+        if (!path || !path_is_absolute(path))
```

Comment:
Hmm? Why? Unexpected paths as input should really be caught as programming errors.